addons/website_profile/static/src/js/website_profile.js

```diff
@@ -25,12 +25,23 @@ publicWidget.registry.websiteProfile = publicWidget.Widget.extend({
     _onSendValidationEmailClick: function (ev) {
         ev.preventDefault();
         var $element = $(ev.currentTarget);
+
+        if (this.emailRequested) {
+            return;
+        }
+
+        $element[0].style.cursor = "progress";
+        this.emailRequested = true;
         this.rpc('/profile/send_validation_email', {
             'redirect_url': $element.data('redirect_url'),
         }).then(function (data) {
             if (data) {
                 window.location = $element.data('redirect_url');
             }
```

Comment:
What happens if `!data`? I'm thinking, do we maybe want to do something like reset the UI/allow clicking again?

Reply:
Looking at the py it seems there are no cases where this happens

Reply:
Hmm, then I'm not sure how to proceed to be honest. Maybe that "if" is a leftover from before some changes where made to the API? Or maybe in some cases we would like to handle a no-error response that still doesn't contain data, i.e. maintenance mode or a server restart etc

However if it exists, it definitely is an edge case. Let me know if you find/think of anything

Reply:
What if we did redirect? In this case, I don't think we want to re-enable the link.

Reply:
> What if we did redirect? In this case, I don't think we want to re-enable the link.

The code will never be executed if we get redirected, I don't see the problem.
If we get redirected nothing will happen => good
Otherwise (error, no data, etc..) we restore the possibility of sending a new email.